chef/automate
# Added API to get server users list from backend DB

### :nut_and_bolt: Description: What code changed, and why?
List the users in a server from back end DB



### :chains: Related Resources
https://github.com/chef/automate/issues/5732

### :+1: Definition of Done
I have added API to list the users in a server from back end DB

### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs

Request

```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/automateinfraserverusers'```

Response

```
{
    "users": [
        {
            "id": "123",
            "server_id": "local-dev",
            "infra_server_username": "Infrauser1",
            "credential_id": "145",
            "connector": "local",
            "automate_user_id": "autouser1",
            "is_server_admin": false
        },
        {
            "id": "321",
            "server_id": "local-dev",
            "infra_server_username": "Infrauser2",
            "credential_id": "147",
            "connector": "local",
            "automate_user_id": "autouser2",
            "is_server_admin": true
        }
    ]
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
